### PR TITLE
Fix extruder motor running in DRYRUN

### PR DESCRIPTION
**Pull request by clefranc**
The extruder motor was running while using the DRYRUN debug function.

### CODE REVIEW

File: Marlin/planner.cpp [hunk 1]
```diff
@@ -502,7 +502,7 @@ float junction_deviation = 0.1;
 
   #ifdef PREVENT_DANGEROUS_EXTRUDE
     if (de) {
-      if (degHotend(extruder) < extrude_min_temp && !(marlin_debug_flags & DEBUG_DRYRUN)) {
+      if (degHotend(extruder) < extrude_min_temp && ((marlin_debug_flags & DEBUG_DRYRUN) != 0)) {
```
Comment (tonnico) [inline Marlin/planner.cpp:510]: `((marlin_debug_flags & DEBUG_DRYRUN) != 0)` == `(marlin_debug_flags & DEBUG_DRYRUN)`

**Comment by clefranc** [comment]
@Wurstnase I know! Weird isn't it? Did you actually tried DRYRUN?

**Comment by AnHardt** [comment]
@MagoKimbra @clefranc 
Yesterday my feeling said, that i want to see the extruder movement. Today i looked up Repetier Documentation.

> In dry run mode, the firmware will ignore all commands to set temperature or extrude.

So indeed if we want to be "Repetier Host compatible" we have to patch.

**Comment by MagoKimbra** [comment]
Uhm if you don't want the mevement extruder in dryryn mode then it is better to delete control, because if dryrun is disabled the extruder not move...
But i want the mevement extruder for totally check printer...

**Comment by tonnico** [comment]
> I know! Weird isn't it? Did you actually tried DRYRUN?

Actually it is `!something` and not `something != 0`
`!a` == `a == 0`
`a` == `a != 0`
Nothing strange i think ;)

**Comment by tonnico** [comment]
@MagoKimbra a dryrun is not a testrun

**Comment by clefranc** [comment]
@AnHardt @MagoKimbra I understand your point of view, as it can be very interesting to see actual extruder movement. In my case, I'm using DRYRUN to test delta movements in **_real**_ conditions. The filament is then always engaged (I'm using an EZStruder) and not easy to disable (remove tension or disconnect motor).

Perhaps we need an option like DISABLE_EXTRUDER_MOTOR_IN_DRYRUN.

**EDIT:** or ENABLE_EXTRUDER_MOTOR_IN_DRYRUN

**Comment by MagoKimbra** [comment]
But DryRun or printing means empty print without filament, the name says it all ...

**Comment by tonnico** [comment]
I like the idea for a dryrun to test a print. But i wouldn't remove the filament for it.

**Comment by MagoKimbra** [comment]
Ok, but better is:

```
if (degHotend(extruder) < extrude_min_temp) {
            position[E_AXIS] = target[E_AXIS]; //behave as if the move really took place, but ignore E part
            de = 0; // no difference
            if (!(debugLevel & DEBUG_DRYRUN)) {
              SERIAL_ECHO_START;
              SERIAL_ECHOLNPGM(MSG_ERR_COLD_EXTRUDE_STOP);
            }
          }
```

**Comment by thinkyhead** [comment]
@MagoKimbra :+1:

**Comment by boelle** [comment]
i will merge this one as soon travis shows green light

why would anyone remove the filament to do a dry print?

**Comment by AnHardt** [comment]
@boelle 
#2248 is the better candidate.

**Comment by boelle** [comment]
ok will close this one then.

have asked for #2248 to be rebased as it cant merge automatic

**Pull request closed without merging**